source/flow_network.py

```diff
@@ -54,9 +65,77 @@ def __init__(self, imp_readnetwork: readnetwork.ReadNetwork, imp_writenetwork: w
         self._imp_buildsystem = imp_buildsystem
         self._imp_solver = imp_solver
         self._imp_rbcvelocity = imp_rbcvelocity
+        self._imp_balance = imp_balance
+        self.imp_iterative = imp_iterative
+
+        # Threshold for zero-loops
+        self.zeroFlowThreshold = None
 
         # "Reference" to parameter dict
         self._PARAMETERS = PARAMETERS
+
+        # Iterative procedure
+        self.alpha = None
+        self.sor = True
+        # the number of iterations performed
+        # 1st (0) is to stabilize the iteration
+        self.iteration = 0
+        # Zero-flow threshold
+        self.zeroFlowThresholdMagnitude = None
+        # Threshold terative approach with our sor
+        self.two_MagnitudeThreshold = None
+        # save residual max and norm over the iteration
+        self.residualOverIterationMax = []
+        self.residualOverIterationNorm = []
+        # to save the value of Alpha's during the iteration and display in the final plot
+        self.alphaSave = [1]
+        # to stop our computation after a certain iteration
+        self.our_convergence_criteria = False
+        # array for discharge hematicrit
+        self.pressure_node = None
+        # assistant variables for an iterative process over 4000 iteration
+        self.vessel_general = None
+        self.vessel_value_hd = None
+        self.vesel_flow_change = None
+        self.vessel_value_flow = None
+        self.node_values_hd = None
+        self.node_values_flow = None
+        self.node_relative_residual = None
+        self.indices_over = None
+        self.indices_over_blue = None
+        self.families_dict_total = None
+        # Convergence criteria for iterative approaches Berg/Rasmussen
+        self.hd_convergence_criteria = None
+        self.flow_convergence_criteria = None
+        self.hd_convergence_criteria_berg = None
+        self.flow_convergence_criteria_berg = None
+        self.pressure_convergence_criteria_berg = None
+        self.berg_criteria = 1e-6
+        self.r_value = 10
+        # Threshold for Rasmussen approach
+        self.rasmussen_hd_threshold = None
+        self.rasmussen_flow_threshold = None
+        # Assistence for iterative approaches of Berg
+        self.inflow = None
+        self.inflow_pressure_node = None
+        self.average_inlet_pressure = []
+        self.bergIteration = []
+        self.Berg1 = []
+        self.Berg2 = []
+        self.BergPressure = []
+        self.BergFlow = []
+        self.BergHD = []
+        self.BergFirstPartEq = []
+        self.BergSecondPartEq = []
+        self.vessel_flow_change_total = None
+        self.maxBalance = None
+        self.node_flow_change_total = None
+        self.vessel_flow_change = None
+        self.node_flow_change = None
+        self.node_residual = None
+        self.positions_of_elements_not_in_boundary = None
+        self.local_balance_rbc = None
+
```

Comment:
Can you double check if you can add more comments next to each variable?
This would be very useful/helpful in case we want to use your test case.
Thanks!